chainer/functions/normalization/batch_normalization.py

```diff
@@ -301,6 +301,12 @@ def forward(self, inputs):
 
     def backward(self, indexes, grad_outputs):
         x, gamma, mean, var = self.get_retained_inputs()
+
+        if not hasattr(self, 'inv_std'):
```

Comment:
Could you move the computation of `inv_var` and `inv_std` to https://github.com/chainer/chainer/pull/3468/files#diff-148edea535bb806260c487ca31869a91R331 and adds `inv_std = None` to https://github.com/chainer/chainer/pull/3468/files#diff-148edea535bb806260c487ca31869a91R234? I feel it's more natural fix (the original code is strange, but such a fix completes the original incomplete implementation).